workspaces/arborist/lib/arborist/build-ideal-tree.js

```diff
@@ -269,6 +269,18 @@ module.exports = cls => class IdealTreeBuilder extends cls {
     this[_complete] = !!options.complete
     this[_preferDedupe] = !!options.preferDedupe
     this[_legacyBundling] = !!options.legacyBundling
+
+    // validates list of update names, they must
+    // be dep names only, no semver ranges are supported
+    const validationError =
+      new TypeError('Update arguments must not contain semver ranges')
```

Comment:
This is likely to be confusing to someone who specifies a semver value and not a range, `@1.3.0` is still gonna error, and while it may technically be a "range" I don't think of it that way.

I can't think of what *should* go here, and if nobody else can we can ship this as-is.  Just a small point, really.

Reply:
How about "must not contain package specifiers"?

Reply:
that's a great point @wraithgar and I do like the wording from @ljharb but now that you made me think of this, maybe we should include a "Did you mean this?" section or something similar?

```
$ npm update foo@bar
Error: Update arguments must not contain package version specifiers

Did you mean this?
    npm update foo
```

Reply:
The existing "did you mean" only works w/ aliases and trying to make it work for this is quite audacious. idk if it's a good use of time.

Reply:
"Must not contain package version specifiers" seems clear enough for the end user to know how to correct the command.

Reply:
> The existing "did you mean" only works w/ aliases and trying to make it work for this is quite audacious. idk if it's a good use of time.

good point 👍 maybe just a multi-line error message with an example though? from an user pov I think an example goes further than just trying to phrase concepts that the user might not understand completely.

So maybe something in the error message itself, in the likes of:

```
$ npm update foo@bar
npm ERR! code EUPDATEARGS
npm ERR! Update arguments must not contain package version specifiers
npm ERR!
npm ERR! Try using the package name instead, e.g:
npm ERR!     npm update foo

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ruyadorno/.npm/_logs/2022-01-24T21_28_42_627Z-debug-0.log
```